src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ReadyToRunGenericHelperNode.cs

```diff
@@ -62,7 +62,7 @@ public static GenericLookupResult GetLookupSignature(NodeFactory factory, ReadyT
                 case ReadyToRunHelperId.VirtualDispatchCell:
                     return factory.GenericLookup.VirtualDispatchCell((MethodDesc)target);
                 case ReadyToRunHelperId.MethodEntry:
-                    return factory.GenericLookup.MethodEntry((MethodDesc)target);
+                    return factory.GenericLookup.MethodEntry((MethodDesc)target, additionalParam != null ? (bool)additionalParam : false);
```

Comment:
Other codegens generate code that obtains address of actual method not unboxing stub for such cases:
```
ldftn      instance string valuetype test.Program/GenStruct`1<!!0>::MakeGenString<!!0>()
calli      instance string ()
```
Unboxing stubs are used in case of delegate construction.